quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/parser/KubeURI.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.kogito.addons.quarkus.k8s.parser;
+
+import java.lang.invoke.MethodHandles;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.ConstraintViolationException;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.ValidatorFactory;
+import javax.validation.constraints.NotNull;
+
+import org.kie.kogito.addons.quarkus.k8s.K8sConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KubeURI {
+
+    private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass().getName());
+
+    @ValidProtocol()
+    private String protocol;
+    @ValidOptionalGVK(message = "the provided ${validatedValue} is invalid or not supported")
+    private GVK gvk;
+    private String namespace;
+    @NotNull(message = "resource name can't be null")
+    private String resourceName;
+    private String rawUrl;
+
+    // KubeURI query parameters
+    private String customPortName;
+    private Map<String, String> customLabel;
+
+    ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
```

Comment:
As you told me yesterday, you can avoid using the validator and just do manual check